tests/unit_tests/repobee/test_git.py

```diff
@@ -200,13 +200,13 @@ async def raise_(pt):
                 "Push failed", 128, b"some error", pt.repo_url
             )
 
-        mocker.patch("_repobee.git._push_async", side_effect=raise_)
+        mocker.patch("_repobee.git.push_async", side_effect=raise_)
```

Comment:
Mocks work by monkeypatching the _reference_ to the object. This is the wrong reference to mock, you need to mock the reference that's used in the module you're testing. It took me a long time to debug the first time I ran into this :)

```suggestion
        mocker.patch("_repobee.git._push.push_async", side_effect=raise_)
```

Reply:
> This is the wrong reference to mock, you need to mock the reference that's used in the module you're testing.

I looked up what is [monkey patching](https://stackoverflow.com/questions/5626193/what-is-monkey-patching) and I think I get the gist and use of it. But I don't think I get why this was the wrong reference and we need to provide an exact reference to the method used. Could you please explain "the reference that's used in the module you're testing" again?